src/strategies/offer_forwarder/PooledForwarder.sol

```diff
@@ -0,0 +1,280 @@
+
+    // check if `this` has enough to satisfy order, otherwise pull _everything_ from aave.
+    if (outTkn.balanceOf(address(this)) < amount) {
+      // we do not verify result - we assume there is now enough, otherwise the order will fail.
+      //TODO aave could be out of funds.
+      AbstractRouter router_ = router();
+
+      //TODO call super instead.
+
+      uint pulled =
+      //reserve should be router because aave does not allow redeeming on behalf of
+       router_.pull({token: outTkn, reserve: address(router_), amount: amount, strict: false /* pull everything */});
+      // return whether some is still missing.
+      return pulled >= amount ? 0 : amount - pulled;
+    }
+
+    // we have decreased owner's balance with the exact amount, so nothing more needed.
+    return 0;
+  }
+
+  function deposit(IERC20 token, uint amount) external {
+    require(
+      TransferLib.transferTokenFrom({
+        token: token,
+        spender: msg.sender, // TODO revisit reserve vs msg.sender - also on Forwarder...
+        recipient: address(this), // TODO consider changing to a direct transfer to router, but would require a deposit function on router, and thus not AbstractRouter
+        amount: amount
+      }),
+      "pooledForwarder/makerTransferFail"
+    );
+    // tokens are now on _this_ and can be pushed to router.
+    // TODO: Should we push to Aave or wait for next order?
+    increaseBalance(token, msg.sender, amount);
+    //TODO: repayThenDeposit is invoked and can pay back debt - consider having the strat be the admin
+    uint pushed = router().push(token, address(router()), amount);
+    require(pushed == amount, "pooledForwarder/pushedWrongAmount");
+  }
+
+  function withdrawToken(IERC20 token, address receiver, uint amount) external returns (bool success) {
+    // update state before calling contracts
+    // if user doesn't have the funds, decreaseBalance will throw
+    decreaseBalance(token, msg.sender, amount);
+
+    // Check local pool first before calling aave
+    uint thisBalance = token.balanceOf(address(this));
+
+    // pull missing from aave into local pool - but only enough - the rest should still generate yield on aave
+    if (thisBalance < amount) {
+      uint pulled = router().pull(token, address(this), amount - thisBalance, true);
+      // this should only happen if Aave is out of liquidity
+      require(pulled + thisBalance == amount, "withdraw/aavePulledWrongAmount");
+    }
+    // Here token.balanceOf(address(this)) >= amount
+
+    // transfer to owner
+    require(TransferLib.transferToken({token: token, recipient: receiver, amount: amount}), "withdraw/transferFailed");
+    return true;
+  }
+
+  function __posthookSuccess__(MgvLib.SingleOrder calldata order, bytes32 makerData)
+    internal
+    override
+    returns (bytes32)
+  {
+    // reposts residual if any (conservative hook)
+    bytes32 repost_status = super.__posthookSuccess__(order, makerData);
```

Comment:
Why define a variable, if we are just ignoring it? Why create it in the first place?